user/topgrade/template.py

```diff
@@ -0,0 +1,31 @@
+pkgname = "topgrade"
+pkgver = "15.0.0"
+pkgrel = 0
+build_style = "cargo"
+hostmakedepends = ["cargo-auditable"]
+pkgdesc = "System updater written in Rust"
+maintainer = "tulilirockz <tulilirockz@outlook.com>"
+license = "GPL-3.0-or-later"
+url = "https://github.com/topgrade-rs/topgrade"
+source = f"{url}/archive/refs/tags/v{pkgver}.tar.gz"
+sha256 = "53c6521041a6ffddf1ccb13f404f131919a2ef48deb3974fc71dc3be08db6cd0"
+
+
+def post_build(self):
+    TOPGRADE_BIN = f"target/{self.profile().triplet}/release/topgrade"
+    with open(f"{self.cwd}/topgrade.bash", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "bash", stdout=o)
+    with open(f"{self.cwd}/topgrade.fish", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "fish", stdout=o)
+    with open(f"{self.cwd}/_topgrade", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-completion", "zsh", stdout=o)
+    with open(f"{self.cwd}/topgrade.8", "w") as o:
+        self.do(TOPGRADE_BIN, "--gen-manpage", stdout=o)
```

Comment:
```suggestion
    for shell in ["bash", "fish", "zsh"]:
        with open(f"{self.cwd}/topgrade.{shell}", "w") as o:
            self.do(f"target/{self.profile().triplet}/release/topgrade", "--gen-completion", "bash", stdout=o)
    with open(f"{self.cwd}/topgrade.8", "w") as o:
        self.do(f"target/{self.profile().triplet}/release/topgrade", "--gen-manpage", stdout=o)
```
also side note, we don't typically use screaming case in cports

Reply:
oh! i thought that ZSH *had* to have the `_` prefix! sorry for the suboptimal implementation

Reply:
it does, but install_completion renames the file